src/client/java/net/neoforged/neoforge/client/event/EntityRenderersEvent.java

```diff
@@ -227,52 +229,61 @@ public EntityRendererProvider.Context getContext() {
         }
 
         /**
-         * Registers the entity model for a skull block with the given {@link SkullBlock.Type}.
+         * Registers the entity model for a skull block with the given {@link SkullBlock.Type}, and optionally registers a skull texture to the {@link SkullBlockRenderer#SKIN_BY_TYPE} map.
          *
          * @param type          a unique skull type; an exception will be thrown if multiple mods register models
          *                      for the same type or a mod tries to register a model for a vanilla type
          * @param layerLocation the key that identifies the {@link LayerDefinition} used by the model
          * @param factory       the factory to create the skull model instance, taking in the root {@link ModelPart} and
          *                      returning the model.
+         * @param skullTexture  the skull texture to put in the {@link SkullBlockRenderer#SKIN_BY_TYPE} map.
          */
-        public void registerSkullModel(SkullBlock.Type type, ModelLayerLocation layerLocation, Function<ModelPart, SkullModelBase> factory) {
-            this.registerSkullModel(type, modelSet -> factory.apply(modelSet.bakeLayer(layerLocation)));
+        public void registerSkullModel(SkullBlock.Type type, ModelLayerLocation layerLocation, Function<ModelPart, SkullModelBase> factory, @Nullable ResourceLocation skullTexture) {
+            this.registerSkullModel(type, modelSet -> factory.apply(modelSet.bakeLayer(layerLocation)), skullTexture);
         }
 
         /**
-         * Registers the entity model for a skull block with the given {@link SkullBlock.Type}.
+         * Registers the entity model for a skull block with the given {@link SkullBlock.Type}, and optionally registers a skull texture to the {@link SkullBlockRenderer#SKIN_BY_TYPE} map.
          *
-         * @param type    a unique skull type; an exception will be thrown if multiple mods register models for
-         *                the same type or a mod tries to register a model for a vanilla type
-         * @param factory the factory to create the skull model instance. A typical implementation will simply bake
-         *                a model using {@link EntityModelSet#bakeLayer(ModelLayerLocation)} and pass it to the
-         *                constructor for {@link SkullModel}
+         * @param type         a unique skull type; an exception will be thrown if multiple mods register models for
+         *                     the same type or a mod tries to register a model for a vanilla type
+         * @param factory      the factory to create the skull model instance. A typical implementation will simply bake
+         *                     a model using {@link EntityModelSet#bakeLayer(ModelLayerLocation)} and pass it to the
+         *                     constructor for {@link SkullModel}
+         * @param skullTexture the skull texture to put in the {@link SkullBlockRenderer#SKIN_BY_TYPE} map.
```

Comment:
```suggestion
         * @param skullTexture the skull texture to put in the {@link SkullBlockRenderer#SKIN_BY_TYPE} map, if provided.
```
The same for the other two methods.